### PR TITLE
[Diagnostics] In DefineMemberBasedOnUse::diagnoseForAmbiguity, use the base type from each solution

### PR DESCRIPTION
**Pull request by hborla**
Previously, the code only used the base type from the first solution. This resulted in potentially calling `solution.simplifyType` on a type that involved type variables that didn't appear in that solution.

Resolves: rdar://problem/61005878

### CODE REVIEW

**Comment by hborla** [comment]
@swift-ci please smoke test

File: lib/Sema/CSFix.h [hunk 1]
```diff
@@ -291,7 +291,10 @@ class ConstraintFix {
   virtual bool diagnose(const Solution &solution,
                         bool asNote = false) const = 0;
 
-  virtual bool diagnoseForAmbiguity(ArrayRef<Solution> solutions) const { return false; }
+  virtual bool diagnoseForAmbiguity(ArrayRef<Solution> solutions,
+                                    ArrayRef<ConstraintFix *> fixes) const {
```
Comment (xedin) [inline lib/Sema/CSFix.h:295]: Could you please adjust this in way that it takes `ArrayRef<std::pair<const Solution &, Constraint Fix *>>`. That would future proof this interface a bit...

**Review by xedin — APPROVED** [review]
LGTM!

**Comment by hborla** [comment]
@swift-ci please smoke test

**Comment by hborla** [comment]
Oh, yeah that's easy, thanks!

**Comment by hborla** [comment]
@swift-ci please smoke test

**Comment by hborla** [comment]
@swift-ci please smoke test OS X platform

**Pull request merged**